python/plot_porkchop.py

```diff
@@ -0,0 +1,176 @@
+Copyright (c) 2016, Enne Hekma (ennehekma@gmail.com)
+All rights reserved.
+'''
+
+# Set up modules and packages
+# Plotting
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from matplotlib import gridspec
+from mpl_toolkits.mplot3d import Axes3D
+from mpl_toolkits.mplot3d import axes3d
+from nlcmap import nlcmap
+from matplotlib import rcParams
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+import math
+
+# I/O
+import commentjson
+import json
+from pprint import pprint
+import sqlite3
+
+# Numerical
+import numpy as np
+import pandas as pd
+
+# System
+import sys
+import time
+
+print ""
+print "------------------------------------------------------------------"
+print "                               D2D                                "
+print "                              0.0.2                               "
+print "      Copyright (c) 2015-2016, K. Kumar (me@kartikkumar.com)      "
+print "      Copyright (c) 2016, E.J. Hekma (ennehekma@gmail.com)        "
+print "------------------------------------------------------------------"
+print ""
+
+# Start timer.
+start_time = time.time( )
+
+print ""
+print "******************************************************************"
+print "                          Input parameters                        "
+print "******************************************************************"
+print ""
+
+# Parse JSON configuration file
+# Raise exception if wrong number of inputs are provided to script
+if len(sys.argv) != 2:
+    raise Exception("Only provide a JSON config file as input!")
+
+json_data = open(sys.argv[1])
+config = commentjson.load(json_data)
+json_data.close()
+pprint(config)
+
+print ""
+print "******************************************************************"
+print "                            Operations                            "
+print "******************************************************************"
+print ""
+
+print "Fetching porkchop plot data from database ..."
+
+# Connect to SQLite database.
+try:
+    database = sqlite3.connect(config['database'])
+
+except sqlite3.Error, e:
+
+    print "Error %s:" % e.args[0]
+    sys.exit(1)
+
+if config['objects']==[]:
+	best = pd.read_sql("SELECT DISTINCT(departure_object_id), arrival_object_id, time_of_flight,  \
+										departure_epoch, transfer_delta_v  						  \
+	                    FROM lambert_scanner_results 											  \
+	 					ORDER BY transfer_delta_v ASC 											  \
+	 					LIMIT 10 ",																  \
+	                    database )
+
+	a = (best['departure_object_id'][0])
+	b = (best['arrival_object_id'][0])
+else:
+	a = config['objects'][0]
+	b = config['objects'][1]
+
+print "Porkchop plot figure being generated for transfer between TLE objects", a, "and", b, "..."
+
+times_of_flight = pd.read_sql_query("	SELECT DISTINCT time_of_flight 							  \
+										FROM lambert_scanner_results",							  \
+										database)
```

Comment:
function arguments not aligned properly, '\' are missing, and extra spaces between " and SELECT to be removed from function argument